src/api/spec/models/kiwi/image_spec.rb

```diff
@@ -76,4 +84,159 @@
       it { expect(subject.valid?).to be_falsey }
     end
   end
+
+  describe '.to_xml' do
+    context 'without a package' do
+      context 'without any repository' do
+        it { expect(kiwi_image.to_xml).to eq(Kiwi::Image::DEFAULT_KIWI_BODY) }
+      end
+
+      context 'with some repositories' do
+        before do
+          kiwi_image.repositories << create(:kiwi_repository)
+        end
+
+        subject { Nokogiri::XML::DocumentFragment.parse(kiwi_image.to_xml) }
+
+        it { expect(subject.errors).to be_empty }
+        it { expect(subject.xpath('.//image').length).to be(1) }
+        it { expect(subject.xpath('.//image/description').length).to be(1) }
+        it { expect(subject.xpath('.//image/packages').length).to be(1) }
+        it { expect(subject.xpath('.//image/repository').length).to be(1) }
+      end
+    end
+
+    context 'without kiwi image file' do
+      after do
+        # We need to be logged to destroy a package
+        login user
+        subject.package.destroy
+        logout
+      end
+
+      subject { create(:kiwi_image_with_package, project: project) }
+
+      it 'returns nil' do
+        expect(subject.to_xml).to be_nil
+      end
+    end
+
+    context 'with kiwi image file' do
+      let(:kiwi_image) { create(:kiwi_image_with_package, project: project, with_kiwi_file: true, file_content: kiwi_xml) }
+
+      after do
+        # We need to be logged to destroy a package
```

Comment:
I think this is obviously, as you are just login the user before destroying the package, and logging it out just after that. And this is also something really normal in our code. I would like to remove this comment.